src/elastio-snap.c

```diff
@@ -4199,8 +4200,12 @@ static MRF_RETURN_TYPE tracing_mrf(struct request_queue *q, struct bio *bio){
 		MRF_SET_RETURN_VALUE(submit_bio_noacct(bio));
 	}
 #else
-	if(orig_mrf) ret = __elastio_snap_call_mrf(orig_mrf, q, bio);
-	else LOG_ERROR(-EFAULT, "error finding original_mrf");
+	if(orig_mrf) {
+		ret = __elastio_snap_call_mrf(orig_mrf, q, bio);
+	} else {
+		LOG_WARN("error finding original_mrf");
+		MRF_SET_RETURN_VALUE(elastio_snap_submit_bio(bio));
```

Comment:
Why then
```
MRF_SET_RETURN_VALUE(elastio_snap_submit_bio(bio));
```
but not
```
MRF_SET_RETURN_VALUE(elastio_snap_null_mrf(bio));
```
?

Reply:
But you'll be never in this `else` on older kernels. `orig_mrf` which is `bdev->queue->make_request_fn` can be empty (NULL) just on the kernel version 5.8.
And for 5.9+ this code is omitted by `USE_BDOPS_SUBMIT_BIO` precompilation directive.

Reply:
It doesn't matter, because in our [suspected] scenario `orig_mrf` couldn't even be initialized because `tracer_matches_bio()` returned `false`. This is why we have that exception.